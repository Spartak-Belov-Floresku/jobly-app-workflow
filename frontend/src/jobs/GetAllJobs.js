import React, { useState, useEffect } from "react";
import SearchForm from "../forms/SearchForm";
import Api from "../dbClass/Api";
import Loader from "../helper/Loader";
import ListOfJobs from "./ListOfJobs";

/** Show page with list of all jobs available.
 *
 * On mount, loads jobs from API.
 * Re-loads filtered jobs on submit from search form.
 *
 * GetAllJobs -> ListOfJobs -> JobDetails
 *
 * This is routed to at /jobs
 */

const GetAllJobs = () => {

  const [jobs, setAllJobs] = useState(null);

  useEffect(function(){
    searchJobs();
  }, []);

  // handle search form submit; get all results that will matched
  const searchJobs = async (title) => {
    let jobs = await Api.getJobs(title);
    setAllJobs(jobs);
  }

  if (!jobs) return <Loader />;

  return (
      <div className="GetAllJobs col-md-8 offset-md-2">
        <SearchForm searchFor={searchJobs} />
        {jobs.length
            ? <ListOfJobs jobs={jobs} />
            : <p className="lead">Sorry, no results were found!</p>
        }
      </div>
  );
}

export default GetAllJobs;