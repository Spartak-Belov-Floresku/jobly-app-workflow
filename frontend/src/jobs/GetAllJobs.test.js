import { render } from "@testing-library/react";
import GetAllJobs from "./GetAllJobs";

it("renders without crashing", () => {
  render(<GetAllJobs />);
});

it("matches snapshot with no jobs", () => {
  const { asFragment } = render(<GetAllJobs />);
  expect(asFragment()).toMatchSnapshot();
});