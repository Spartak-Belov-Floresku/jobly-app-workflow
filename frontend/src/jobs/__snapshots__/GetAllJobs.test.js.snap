// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <div
    class="SpinnerLoad"
  >
    Loading ...
  </div>
</DocumentFragment>
`;
